arkouda/join.py

```diff
@@ -241,10 +273,40 @@ def inner_join(
         keep12 = keep
     else:
         if whereargs is not None:
-            # Gather right whereargs
-            rightWhere = whereargs[1][byRight.permutation][ranges]
-            # Expand left whereargs
-            leftWhere = broadcast(fullSegs, whereargs[0][keep], ranges.size)
+            if t == Categorical.objtype and isinstance(cats, Strings):
+                # Find the corresponding codes value for each term in the whereargs
+                repMsg = generic_msg(
+                    cmd="codeMapping",
+                    args={
+                        "categories": cats,
+                        "query": whereargs[0][keep],  # type: ignore
+                    },
+                )
+                lCodes = create_pdarray(repMsg)
```

Comment:
Running the same example i put in my original review gives me a different error on this call
```python
>>> ak.join.inner_join(str_left, str_right, wherefunc = where_func, whereargs=(int_left,int_right))
---------------------------------------------------------------------------
RuntimeError: Error: Unhandled SymbolEntryType PrimitiveTypedArraySymEntry
```

Reply:
Fixed